syntax = "proto3";

package NFMsg; 

enum EServerState
{
	EST_CRASH = 0;
	EST_NARMAL = 1;
	EST_BUSY = 2;
	EST_FIRE = 3;
	EST_MAINTEN = 4;
}

message ServerInfoReport
{
	int32   server_id = 1;
	int32   server_type = 2;
	string  server_name = 3;
	string  server_ip = 4;
	int32   server_port = 5;
	int32   server_http_port = 6;
	int32   server_max_online = 7;
	int32   server_cur_online = 8;
	uint32  server_state = 9;
	string  system_info = 10;
	uint64  total_mem = 11;
	uint64  free_mem = 12;
	uint64  used_mem = 13;
	double  proc_cpu = 14;	//cur process cpu used
	uint64  proc_mem = 15;	//cur process mem used
	uint32  proc_thread = 16; //cur process thread num
	string  proc_name = 17;
	string  proc_cwd = 18; 
	uint32  proc_pid = 19;
	string  last_date = 20;
	uint64  last_time = 21;
}

message ServerInfoReportList
{
	repeated ServerInfoReport server_list = 1;
}

message ServerErrorLogMsg
{
	string error_log = 1;
	string func_log = 2;
	uint64 player_id = 3;
	string server_name = 4;
}